src/ecs/commands/runCommandInContainer.ts

```diff
@@ -0,0 +1,162 @@
+        ),
+        placeHolder: localize('AWS.command.ecs.runCommandInContainer.placeHolder', 'Command to run'),
+        ignoreFocusOut: true,
+    })
+    if (!command) {
+        return
+    }
+
+    // Warn the user that commands may modify the container, give the option to dismiss future prompts
+    const configuration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
+    if (configuration.readSetting('ecs.warnRunCommand')) {
+        const choice = await window.showInformationMessage(
+            localize(
+                'AWS.command.ecs.runCommandInContainer.warnBeforeExecute',
+                'Command may modify the running container {0}. Are you sure?',
+                node.continerName
+            ),
+            { modal: true },
+            localize('AWS.generic.response.yes', 'Yes'),
+            localize('AWS.message.prompt.yesDontAskAgain', "Yes, and don't ask again")
+        )
+        if (choice === undefined) {
+            getLogger().debug('ecs: Cancelled runCommandInContainer')
+            return
+        } else if (choice === localize('AWS.message.prompt.yesDontAskAgain', "Yes, and don't ask again")) {
+            configuration.writeSetting('ecs.warnRunCommand', false, vscode.ConfigurationTarget.Global)
+        }
+    }
+
+    const response = await new ChildProcess(
+        true,
+        'aws',
+        undefined,
+        'ecs',
+        'execute-command',
+        '--cluster',
+        node.clusterArn,
+        '--task',
+        task,
+        '--container',
+        node.continerName,
+        '--command',
+        command,
+        '--interactive'
+    ).run()
+
+    if (response.exitCode !== 0) {
+        getLogger().error(response.stderr)
+        recordEcsRunExecuteCommand({ result: 'Failed', ecsExecuteCommandType: 'command' })
+    } else {
+        showOutputMessage(response.stdout, outputChannel)
+        recordEcsRunExecuteCommand({ result: 'Succeeded', ecsExecuteCommandType: 'command' })
+    }
+}
+
+async function verifyCliAndPlugin(window: Window): Promise<void> {
+    const verifyAwsCliResponse = await new ChildProcess(true, 'aws', undefined, '--version').run()
+    if (verifyAwsCliResponse.exitCode !== 0) {
+        const noCli = localize(
+            'AWS.command.ecs.runCommandInContainer.noCliFound',
+            'Please install the {0} CLI before proceding',
+            getIdeProperties().company
+        )
+        window.showErrorMessage(noCli)
+        throw new Error('The AWS CLI is not installed.')
+    }
+
+    const verifySsmPluginResponse = await new ChildProcess(true, 'session-manager-plugin').run()
+    if (verifySsmPluginResponse.exitCode !== 0) {
+        window.showErrorMessage(
+            localize(
+                'AWS.command.ecs.runCommandInContainer.noPluginFound',
+                'Please install the SSM Session Manager plugin before proceding'
+            )
+        )
+        throw new Error('The SSM Session Manager plugin is not installed.')
```

Comment:
Why throw an error if we are using showErrorMessage()? Does this manifest as multiple messages?

Reply:
Error not really needed here.  Changed so that `runCommandInContainer `will simply return after displaying the message.